spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/DeleteOrphanFilesSparkAction.java

```diff
@@ -161,6 +163,12 @@ public DeleteOrphanFilesSparkAction olderThan(long newOlderThanTimestamp) {
     return this;
   }
 
+  @Override
+  public DeleteOrphanFilesSparkAction batchSize(int batchSize) {
```

Comment:
You probably want to use `Preconditions` to validate and throw `IllegalArgumentException ` for invalid batch size such as 0 or negative number based on the javadoc in `api/src/main/java/org/apache/iceberg/actions/DeleteOrphanFiles.java` interface

Reply:
Great catch, I'm removing this in the spark procedure but it should be done at the file IO layer, which I missed

Reply:
Actually, nvm I already have this check here for the S3FileIO case: https://github.com/apache/iceberg/blob/master/aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java#L527